Assignment4.java

```diff
@@ -0,0 +1,87 @@
+package assignment4;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class Assignment4 {
+
+    @Test
+    public void testNoIslands() {
+        boolean[][] grid = new boolean[][] {
+                {false, false, false},
+                {false, false, false}
+        };
+        int result = new IslandCounter(grid).countIslands();
+        assertEquals("Found some islands", 0, result);
+    }
+
+    @Test
+    public void testSingleCellIsland() {
+        boolean[][] grid1 = new boolean[][] {
+                {true,  false, false},
+                {false, false, false}
+        };

```

Comment:
Really good set of test coverage here. 2 extra that I might have added would have been:

boolean[][] grid1 = new boolean[][] {
                {false, false}
                {true,  false},
                 {false, false}
         };

and 
boolean[][] grid1 = new boolean[][] {
                {false, false}
                {false, true},
                 {false, false}
         };
as mistakes are often made on edges in this kind of puzzle.